src/acc/libsmm_acc/CMakeLists.txt

```diff
@@ -46,30 +46,22 @@ if (USE_CUDA)
 
 else () # i.e. USE_HIP
 
-  set_source_files_properties(${LIBSMM_ACC_SRC_FILES}
-                              PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-  # hip_add_library does not support OBJECT libraries. For this reason, we are
-  # forced to use a static library
-  hip_add_library(libsmm_acc STATIC ${LIBSMM_ACC_FILES})
-  target_include_directories(libsmm_acc PRIVATE ${HIP_PATH}/../include)
-  target_link_libraries(libsmm_acc INTERFACE "stdc++")
-  target_link_libraries(libsmm_acc PUBLIC ${HIPBLAS})
+  add_library(libsmm_acc SHARED ${LIBSMM_ACC_FILES})
+  target_link_libraries(libsmm_acc PUBLIC roc::hipblas hip::host)
+
```

Comment:
similar here: `OBJECT` possible and would it make sense?

Reply:
I've removed the object files, these are source files of 1 lib anyways